tests/integration/cartography/intel/aws/ec2/test_ec2_volumes.py

```diff
@@ -3,123 +3,147 @@
+from cartography.intel.aws.ec2.volumes import sync_ebs_volumes
 from tests.data.aws.ec2.instances import DESCRIBE_INSTANCES
-
+from tests.data.aws.ec2.volumes import DESCRIBE_VOLUMES
+from tests.integration.cartography.intel.aws.common import create_test_account
+from tests.integration.util import check_nodes
+from tests.integration.util import check_rels
 
 TEST_ACCOUNT_ID = '000000000000'
 TEST_REGION = 'eu-west-1'
 TEST_UPDATE_TAG = 123456789
 
 
-def test_load_volumes(neo4j_session):
+@patch.object(cartography.intel.aws.ec2.volumes, 'get_volumes', return_value=DESCRIBE_VOLUMES)
+def test_sync_ebs_volumes(mock_get_vols, neo4j_session):
     # Arrange
-    data = tests.data.aws.ec2.volumes.DESCRIBE_VOLUMES
-    transformed_data = cartography.intel.aws.ec2.volumes.transform_volumes(data, TEST_REGION, TEST_ACCOUNT_ID)
+    boto3_session = MagicMock()
+    create_test_account(neo4j_session, TEST_ACCOUNT_ID, TEST_UPDATE_TAG)
 
     # Act
-    cartography.intel.aws.ec2.volumes.load_volumes(
+    sync_ebs_volumes(
         neo4j_session,
-        transformed_data,
-        TEST_REGION,
+        boto3_session,
+        [TEST_REGION],
         TEST_ACCOUNT_ID,
         TEST_UPDATE_TAG,
+        {'UPDATE_TAG': TEST_UPDATE_TAG, 'AWS_ID': TEST_ACCOUNT_ID},
     )
 
     # Assert
-    expected_nodes = {
-        "v-01", "v-02",
+    assert check_nodes(neo4j_session, 'EBSVolume', ['arn']) == {
+        ('arn:aws:ec2:eu-west-1:000000000000:volume/vol-03',),
+        ('arn:aws:ec2:eu-west-1:000000000000:volume/vol-0df',),
     }
 
-    nodes = neo4j_session.run(
-        """
-        MATCH (r:EBSVolume) RETURN r.id;
-        """,
-    )
-    actual_nodes = {n['r.id'] for n in nodes}
-
-    assert actual_nodes == expected_nodes
-
-
-def test_load_volume_to_account_rels(neo4j_session):
+    # Assert
+    assert check_rels(
+        neo4j_session,
+        'AWSAccount',
+        'id',
+        'EBSVolume',
+        'volumeid',
+        'RESOURCE',
+        rel_direction_right=True,
+    ) == {
+        (TEST_ACCOUNT_ID, 'vol-03'),
+        (TEST_ACCOUNT_ID, 'vol-0df'),
+    }
 
-    # Arrange: Create Test AWSAccount
-    neo4j_session.run(
-        """
-        MERGE (aws:AWSAccount{id: $aws_account_id})
-        ON CREATE SET aws.firstseen = timestamp()
-        SET aws.lastupdated = $aws_update_tag
-        """,
-        aws_account_id=TEST_ACCOUNT_ID,
-        aws_update_tag=TEST_UPDATE_TAG,
-    )
 
-    # Act: Load Test Volumes
-    data = tests.data.aws.ec2.volumes.DESCRIBE_VOLUMES
-    transformed_data = cartography.intel.aws.ec2.volumes.transform_volumes(data, TEST_REGION, TEST_ACCOUNT_ID)
+@patch.object(cartography.intel.aws.ec2.instances, 'get_ec2_instances', return_value=DESCRIBE_INSTANCES['Reservations'])
+@patch.object(cartography.intel.aws.ec2.volumes, 'get_volumes', return_value=DESCRIBE_VOLUMES)
+def test_sync_ebs_volumes_e2e(mock_get_vols, mock_get_instances, neo4j_session):
+    # Arrange
+    neo4j_session.run('MATCH (n) DETACH DELETE n;')
```

Comment:
We shouldn't need this line, because the fixture should automatically do this.

Reply:
https://github.com/lyft/cartography/blob/48f50caed6ac889225e94611b1315f6a95c19470/tests/integration/conftest.py#L12

fixture scope is set to module so it doesn't actually haha. If we change it to 'function' then it'll break all the other tests

Reply:
Alright